### PR TITLE
Two property printing fixes

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by boucman** [comment]
is [not set] parsable by systemd ? wouldnt `# not set` be a bit better both for shell integration and unit-file integration ?

**Comment by keszybz** [comment]
We already have many properties which use `[not set]`.

**Comment by boucman** [comment]
ah, ok. fair enough

File: src/shared/bus-util.c [hunk 1]
```diff
@@ -809,7 +809,17 @@ int bus_print_property(const char *name, sd_bus_message *property, bool value, b
 
                 if (strstr(name, "UMask") || strstr(name, "Mode"))
                         print_property(name, "%04o", u);
-                else
+                else if (streq(name, "UID")) {
+                        if (u == UID_INVALID)
```
Comment (poettering) [inline src/shared/bus-util.c:813]: hmm uid_is_invalid() is the better checker for this I figure

File: src/shared/bus-util.c [hunk 2]
```diff
@@ -809,7 +809,17 @@ int bus_print_property(const char *name, sd_bus_message *property, bool value, b
...
+                                print_property(name, "%s", "[not set]");
+                        else
+                                print_property(name, "%"PRIu32, u);
+                } else if (streq(name, "GID")) {
+                        if (u == GID_INVALID)
```
Comment (poettering) [inline src/shared/bus-util.c:818]: gid_is_valid()

File: src/shared/bus-util.c (hunk 1 above)
Comment (keszybz) [reply src/shared/bus-util.c:813]: Hmmm, when I see `u == <constant>` then I know that the `<constant>` is the special value used as a placeholder. But `uid_is_valid()` is opaque, and feels more like a validation function. We should never allow uid to be equal to 0xFFFF, but if there was an error in our code, we'd want to report that as 0xFFFF, not [unset]. So... I can change it, if you think that's better, but it doesn't feel that way for me.

**Pull request merged**